src/conversations/Conversations.ts

```diff
@@ -74,12 +78,76 @@ export default class Conversations {
 
     return Stream.create<Conversation>(
       this.client,
-      buildUserIntroTopic(this.client.address),
+      [buildUserIntroTopic(this.client.address)],
       messageTransformer,
       filter
     )
   }
 
+  /**
+   * Returns a stream for all new messages from all existing and new conversations.
+   */
+  async streamAllMessages(): Promise<Stream<Message>> {
+    const conversations = await this.list()
+    const dmAddresses: string[] = conversations.map(
+      (conversation) => conversation.peerAddress
+    )
+    const introTopic = buildUserIntroTopic(this.client.address)
+    const topics = this.buildTopicsForAllMessages(dmAddresses, introTopic)
+
+    // If we detect a new intro topic, update the stream's direct message topics to include the new topic
+    const contentTopicUpdater = (msg: Message): string[] | undefined => {
+      if (msg.contentTopic !== introTopic || !messageHasHeaders(msg)) {
+        return undefined
+      }
+      const peerAddress = this.getPeerAddress(msg)
+      if (
+        dmAddresses.includes(peerAddress) ||
+        peerAddress === this.client.address
+      ) {
+        // No need to update if we're already subscribed
+        return undefined
+      }
+      dmAddresses.push(this.getPeerAddress(msg))
+      return this.buildTopicsForAllMessages(dmAddresses, introTopic)
+    }
+
+    // Filter intro topics if already streaming direct messages for that address to avoid duplicates
+    const filter = (msg: Message): boolean => {
```

Comment:
Can't we filter all introTopic messages? IIRC the intro message should be duplicated into the dm topic as well, no?

Reply:
I don't think it's possible to catch new messages coming from new conversations because we aren't subscribed to that DM topic yet (since it didn't exist when the stream started). From my testing, if a fresh convo is kicked off, the intro topic is the only message received and then we subscribe to the DM topic after that. We only protect against dups here in case the convo existed before the intro topic was sent and we were already subscribing to the DM. Lmk if anything seems off with that though!

Reply:
Ah, I see, got it now. 👍

Reply:
That feels a bit racy though, since there may be more messages in the new convo topic. Prudent thing would probably be to list the new conversation after subscribing to make sure we didn't drop anything.

Reply:
Then again, we might fetch some old messages if we list and old dm topic. Hm, not sure if we can actually do much better than you're already doing.

Reply:
>That feels a bit racy though, since there may be more messages in the new convo topic.

I _think_ this should be covered by us creating and appending a new DM topic for the new address as we receive the intro message and including that in the new subscription (still within the new message callback for the intro message):

https://github.com/xmtp/xmtp-js/blob/0bf89e9081b2eb374b333e485fbc0cc01c75ef03/src/Stream.ts#L78

I could absolutely be missing something though so please lmk if I'm misunderstanding!

Reply:
My point was that I don't think there's anything preventing the sender from sending multiple messages before the recipient subscribes to the new conversation (in fact could be even before the recipient receives the intro message). So in theory we could end up showing the very first message (from the intro topic), then miss a few that arrived to the dm topic before we subscribed to it, and then start showing whatever comes after. That's why I suggested to list the topic first. But if the two parties communicated before then it could pick up messages that shouldn't be part of this stream. So I'm not sure if we can make this really bullet proof.

Reply:
I'd say don't let this stop you from merging this. We can ponder the tricky stuff later if it ever becomes relevant.

Reply:
Ahhh got it. Appreciate the explanation @mkobetic 🙏 Will keep an eye out for feedback here for sure.